File: src/Domain/Library/Manuscript.cs
using SharedKernel;

namespace Domain.Library;

public sealed class Manuscript : Entity
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public int WritingPlaceId { get; set; }
    public string Year { get; set; } = string.Empty;
    public int PageCount { get; set; }
    public string Size { get; set; } = string.Empty;
    public int GapId { get; set; }
    public string VersionNo { get; set; } = string.Empty;
    public int LanguageId { get; set; }
    public int SubjectId { get; set; }

    public Language? Language { get; set; }
    public Subject? Subject { get; set; }
    public City? City { get; set; }
    public Gap? Gap { get; set; }
}

File: src/Web.UI/src/app/models/manuscript.ts
export type Manuscript = {
  id: number;
  name: string;
  author: string;
  writingPlaceId: number;
  writingPlaceName?: string;
  year: string;
  pageCount: number;
  size: string;
  gapId: number;
  gapTitle?: string;
  versionNo: string;
  languageId: number;
  languageName?: string;
  subjectId: number;
  subjectTitle?: string;
};

export type CreateManuscriptRequest = {
  name: string;
  author: string;
  writingPlaceId: number;
  year: string;
  pageCount: number;
  size: string;
  gapId: number;
  versionNo: string;
  languageId: number;
  subjectId: number;
};

export type UpdateManuscriptRequest = CreateManuscriptRequest & {
  id: number;
};

export type ManuscriptSearchRequest = {
  name?: string;
  author?: string;
  gapId?: number;
  writingPlaceId?: number;
  languageId?: number;
  subjectId?: number;
  year?: string;
};

export type ManuscriptSearchResponse = {
  id: number;
  name: string;
  author: string;
  gapTitle?: string;
  writingPlaceName?: string;
  year: string;
  languageName?: string;
  subjectTitle?: string;
  pageCount: number;
  size: string;
};


